script/test

```diff
@@ -14,4 +14,5 @@ echo "==> Running ktlint..."
 echo "==> Running tests..."
 ./gradlew test
 
-
+echo "==> Running static code analyser..."
+./gradlew detekt
```

Comment:
I feel like we should run this first, so we fail fast on any detekt issues before running the whole test suite

Reply:
I considered this too. Running it latter has the advantage of getting test feedback as you go without getting interrupted by style or naming nitpicking, especially as we code we might be aware of these problems but are immediately interested in test feedback.

How do other api devs run the tests? Given they take so long I suppose they are running the unit tests outside of script/test for faster feedback? In which case yes adding it to the top here feels right.

Reply:
To be honest, I tend to run the tests in IntelliJ, and only run the full suite in CI, so very rarely use this script

Reply:
From a quick poll it seems that this isn't used regularly so the ordering arguable doesn't matter much. Let's keep it grouped with linting that happens up front as you suggest 👍